model/metrics/vcs.yaml

```diff
@@ -0,0 +1,150 @@
+groups:
+  - id: metric.vcs.repository.change.count
+    type: metric
+    metric_name: vcs.repository.change.count
+    brief: 'The number of changes (pull requests) in a repository, categorized by their state (either open or merged)'
+    instrument: gauge
```

Comment:
it's probably interesting to have two separate metrics - one is `updadowncounter` for number of open PRs and another one (counter or maybe up-down-too since I can reopen PRs) for closed PRs depending on the state (merged or not merged)

Otherwise we're having very different scales and different types of things we measure (almost monotonically incrementing and eventually big number of closed/merged PRs vs probably very small and constant (or so) number of open ones).

Reply:
After discussion in the CICD working group, we think this would be best served by a histogram and will make that change pending any objections.

Reply:
After some further thought and reading [Instrument selection](https://opentelemetry.io/docs/specs/otel/metrics/supplementary-guidelines/#instrument-selection), `updowncounter` seems the best instrument for `vcs.repository.ref.count` and `vcs.repository.change.count` metrics since the measurements are additive over the attributes.
I made this change in aab9f90b5944da890db2c82a00871a0d1430bf44.

Reply:
@lmolkova Why is it an issue if the metric has vastly different values for the different values of the `vcs.change.state` attribute (eg. `open`, `closed`, `merged`)?
We can filter by this attribute if we want to consider open vs closed/merged PRs separately.

Reply:
@lmolkova any thoughts?

Reply:
it could be a problem when you show metric grouped by dimension and one value is so big that the variations in the other one are not visible.

E.g. 

<img width="387" alt="image" src="https://github.com/user-attachments/assets/25b63115-85f0-4a26-aa0b-cba1bb8b15aa">

It probably just means that dashboards should filter on certain values instead of grouping.

In any case, let's not block on this - we can change it if it becomes an issue.